dot/rpc/modules/state.go

```diff
@@ -176,30 +202,81 @@ func (sm *StateModule) GetRuntimeVersion(r *http.Request, req *StateBlockHashQue
 	return err
 }
 
-// GetStorage isn't implemented properly yet.
-func (sm *StateModule) GetStorage(r *http.Request, req *StateStorageQueryRequest, res *StateStorageDataResponse) {
+// GetStorage Returns a storage entry at a specific block's state. If not block hash is provided, the latest value is returned.
+func (sm *StateModule) GetStorage(r *http.Request, req *[]string, res *interface{}) error {
+	// TODO implement handling of block hash parameter (See issue #834)
+	pReq := *req
+	reqBytes, _ := common.HexToBytes(pReq[0]) // no need to catch error here
+
+	item, err := sm.storageAPI.GetStorage(reqBytes)
+	if err != nil {
+		return err
+	}
+
+	if len(item) > 0 {
+		*res = "0x" + hex.EncodeToString(item)
+	} else {
+		*res = nil
+	}
+
+	return nil
 }
 
 // GetStorageHash isn't implemented properly yet.
-func (sm *StateModule) GetStorageHash(r *http.Request, req *StateStorageQueryRequest, res *StateStorageHashResponse) {
+func (sm *StateModule) GetStorageHash(r *http.Request, req *[]string, res *interface{}) error {
```

Comment:
what is this supposed to do?

Reply:
updated comment

Reply:
cool, could you add a TODO for this saying it can be finished when we have the storage changes trie?

Reply:
Done.